tests/system/src/test/java/test/robot/javafx/scene/NodeInitializationBackgroundThreadTest.java

```diff
@@ -0,0 +1,906 @@
+            TreeItem<String> root = new TreeItem<>(null);
+            return root;
+        };
+
+        test(() -> {
+            TreeView<String> c = new TreeView<>();
+            c.setSkin(new TreeViewSkin(c));
+            c.setRoot(createRoot());
+            return c;
+        }, (c) -> {
+            c.setRoot(createRoot());
+            accessControl(c);
+        });
+    }
+
+    private void accessTextInputControl(TextInputControl c) {
+        accessControl(c);
+        c.setPromptText("yo");
+        c.setText(nextString());
+        c.prefHeight(-1);
+    }
+
+    private void accessChart(Chart c) {
+        String title = c.getClass().getSimpleName();
+        c.setTitle(title);
+        c.setAnimated(true);
+        accessRegion(c);
+    }
+
+    private void accessControl(Control c) {
+        accessRegion(c);
+        c.getCssMetaData();
+    }
+
+    private void accessNode(Node c) {
+        c.setFocusTraversable(true);
+        c.requestFocus();
+        c.toFront();
+    }
+
+    private void accessRegion(Region c) {
+        accessNode(c);
+        c.prefHeight(-1);
+        c.prefWidth(-1);
+        c.setPrefWidth(20);
+        c.setPrefHeight(20);
+    }
+
+    private <T extends Node> void test(Supplier<T> generator, Consumer<T> operation) {
+        T visibleNode = generator.get();
+        String title = visibleNode.getClass().getSimpleName();
+
+        setTitle(title);
+        setContent(visibleNode);
+
+        int threadCount = 1 + Runtime.getRuntime().availableProcessors() * 2;
+        AtomicBoolean running = new AtomicBoolean(true);
+        CountDownLatch counter = new CountDownLatch(threadCount);
+
+        try {
+            for (int i = 0; i < threadCount; i++) {
+                new Thread(title) {
+                    @Override
+                    public void run() {
+                        try {
+                            T n = generator.get();
+                            int count = 0;
+                            while (running.get()) {
+                                operation.accept(n);
+
+                                count++;
+                                if ((count % 100) == 0) {
+                                    inFx(() -> {
+                                        operation.accept(visibleNode);
+                                    });
+                                }
+                            }
+                        } finally {
+                            counter.countDown();
+                        }
+                    }
+                }.start();
+            }
+
+            sleep(DURATION);
+        } finally {
+            running.set(false);
+        }
+
+        // let all threads finish
+        try {
+            counter.await(500, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            fail(e);
+        }
+    }
+
+    private static boolean nextBoolean() {
+        return new Random().nextBoolean();
```

Comment:
It is a best practice when using Random to share a single instance and not allocate a new one each time. It is more performant and more random.

Reply:
while true, the shared instance will add synchronization and this is something I would like to avoid.  The degree of randomness is not that important in this case.

the only thing that we lose is reproducibility, but in the case of a headful test it's probably impossible anyway, due to many other factors.

In any case, I think the main purpose of this test is a sort of sanity check of the requirement, not an exhaustive test of all the properties and all the methods that the application can call from a background thread - at least that what I thought at the time of writing.  We can always add more stress to it.

Reply:
OK, that sounds reasonable. Maybe one other thing to consider would be `ThreadLocalRandom`, but it's probably OK ei ther way.